var/spack/repos/builtin/packages/py-anuga/package.py

```diff
@@ -13,20 +15,55 @@ class PyAnuga(PythonPackage):
 
     variant('mpi', default=True, description='Install anuga_parallel')
 
-    # At present AnuGA has only been run and tested using python 2.x.
-    # We recommend python 2.7.
-    depends_on('python@2.6:2.8', type=('build', 'run'))
-    depends_on('py-setuptools@:44', type='build')
-    depends_on('py-numpy@:1.16', type=('build', 'run'))
+    # Non-versioned dependencies for Anuga main and future versions based on python@3.5:
+    depends_on('python@3.5:',            type=('build', 'run'), when='@2.2:')
+    depends_on('gdal+geos+python',       type=('build', 'run'), when='@2.2:')
+    depends_on('py-future',              type=('build', 'run'), when='@2.2:')
+    depends_on('py-matplotlib',          type=('build', 'run'), when='@2.2:')
+    depends_on('py-numpy',               type=('build', 'run'), when='@2.2:')
+    depends_on('py-setuptools',          type=('build'),        when='@2.2:')
+    # Replaces pyar in python3 anuga:
+    depends_on('py-mpi4py',              type=('build', 'run'), when='@2.2:+mpi')
+    depends_on('py-dill',                type=('test'))
+    depends_on('py-nose',                type=('test'))
+    depends_on('py-scipy',               type=('test'))
+    depends_on('py-triangle',            type=('test'))
+
+    # Version-restricted dependencies for anuga@:2.1, converted to use when='@:2.1':
+    depends_on('python@2.6:2.7.18',      type=('build', 'run'), when='@:2.1')
+    depends_on('gdal@:3.2+geos+python',  type=('build', 'run'), when='@:2.1')
+    depends_on('py-matplotlib@:2',       type=('build', 'run'), when='@:2.1')
+    depends_on('py-numpy@:1.16',         type=('build', 'run'), when='@:2.1')
+    depends_on('py-setuptools@:44',      type=('build'),        when='@:2.1')
+    # Workaround for problem in the original concretizer selecting 3.6, wanting python3:
+    depends_on('py-setuptools-scm@:3.5', type=('build'),        when='@:2.1')
+    # pypar is not updated for python3, for python3, py-mpi4py is unsed instead:
+    depends_on('py-pypar',               type=('build', 'run'), when='@:2.1+mpi')
+
+    # Unversioned dependencies of the python2 and python3-based versions
+    depends_on('py-cython',              type=('build'))
     depends_on('py-netcdf4', type=('build', 'run'))
-    depends_on('py-matplotlib@:2', type=('build', 'run'))
-    depends_on('gdal@:3.2+python', type=('build', 'run'))
-    depends_on('py-pypar', when='+mpi', type=('build', 'run'))
 
     # https://github.com/GeoscienceAustralia/anuga_core/issues/247
     conflicts('%apple-clang@12:')
+
+    def patch(self):
+        # MPI tests can only be activated when=+mpi and mpiexec is added to the PATH
+        shutil.rmtree('anuga/parallel/tests')
+        # AttributeError: 'NoneType' object has no attribute 'Intersection'
+        shutil.rmtree('anuga/utilities/tests')
+        # These use csv files from anuga/utilities/tests/data
+        shutil.rmtree('anuga/geometry/tests')
```

Comment:
It might be better to report these issues upstream and hope that someone fixes them in main

Reply:
One of the last fixes fixed the remaining tests: The testsuite is fully fixed now!

Either it was some dependency or adding the geos variant to the existing python variant of gcal or maybe some side effect of adding mpiexec to the path. At least, it runs fine now.